pallets/api/src/fungibles/tests.rs

```diff
@@ -1,32 +1,67 @@
 use codec::Encode;
 use frame_support::{
-	assert_ok,
-	sp_runtime::traits::Zero,
+	assert_noop, assert_ok,
+	dispatch::WithPostDispatchInfo,
+	sp_runtime::{traits::Zero, DispatchError::BadOrigin},
 	traits::fungibles::{
 		approvals::Inspect as ApprovalInspect, metadata::Inspect as MetadataInspect, Inspect,
 	},
 };
 
-use crate::{fungibles::Read::*, mock::*, Read};
+use crate::{
+	fungibles::{
+		weights::WeightInfo as WeightInfoTrait, AssetsInstanceOf, AssetsWeightInfoOf,
+		AssetsWeightInfoTrait, Config, Read::*, ReadResult,
+	},
+	mock::*,
+	Read,
+};
 
 const TOKEN: u32 = 42;
 
+type AssetsError = pallet_assets::Error<Test, AssetsInstanceOf<Test>>;
+type AssetsWeightInfo = AssetsWeightInfoOf<Test>;
 type Event = crate::fungibles::Event<Test>;
+type WeightInfo = <Test as Config>::WeightInfo;
+
+#[test]
+fn encoding_read_result_works() {
+	use crate::fungibles::ReadResult::*;
+	let value = 100 * UNIT;
```

Comment:
Can we define all variables at the top. So, instead of using `value` for the names, use `let amount`, `let name`, `let exists`, and define them at the top.

Reply:
(see the entire test to understand, it redefines `value` a few times)

Reply:
I think a better approach is to separate them out into distinct tests personally.